core/templates/components/graph-services/graph-layout.service.ts

```diff
@@ -493,12 +523,10 @@ export class StateGraphLayoutService {
     };
   }
 
-  // TODO(#7165): Replace 'any' with the exact type. This has been kept as
-  // 'any' because 'nodeData' is a complex dict whose exact type needs to
-  // be researched. Same goes for 'nodeLinks' and the return type.
-  getAugmentedLinks(nodeData: any, nodeLinks: any): any {
+  getAugmentedLinks(
+      nodeData: INodeDataDict, nodeLinks: IGraphLink[]): IAugmentedLink[] {
     var links = cloneDeep(nodeLinks);
-    var augmentedLinks = links.map((link: any) => {
+    var augmentedLinks: IAugmentedLink[] = links.map(link => {
```

Comment:
Do we need to add a type for the callback parameter for map like before?

Reply:
the type is automatically inferred because the type of `links` is known.

Reply:
cool, thanks for explaining!